cli/azd/pkg/output/ux/action_result.go

```diff
@@ -18,8 +19,24 @@ type ActionResult struct {
 
 func (ar *ActionResult) ToString(currentIndentation string) (result string) {
 	if ar.Err != nil {
-		return output.WithErrorFormat("\n%s: %s", "ERROR", ar.Err.Error())
```

Comment:
My proposal:
- create a constructor for `ActionResult` as `NewActionResult(successMessage, followUp string, err error)` 
- Make the fields from ActionResult private (this is really optional)
- Create a new ux type for Error (maybe ux.Error) and implement ToString() and MarshalJSON()
  - Add a constructor for this new ux.Error as `NewUxError(err error)`
- Make NewActionResult() to check if err != nil, and if yes, create a ux.Error inside the ActionResult
- Make ActionResult methods to be a wrapper of ux.Error when it was created with an error.  Otherwise, use it as ActionResult.

This strategy would allow us to produce ux.Error directly if we need it (without depending on an ActionResult) and would also automatically integrate with any actionResult which contains an error.

Reply:
If that sounds like a good idea, I can push a commit to this PR.  just let me know.

Reply:
Can you help clarify on the goals of the proposal? Is it:
1. To reshare implementation for error formatting
2. To decouple ActionResult from the error
3. To provide the opportunity for ux.Error to perform `MarshalJson`?

Reply:
I like the concept of a `UxError` that provides a better UX for visualizing our current errors but we would need to figure out when/where to use this.  One idea is that we should expect our commands to wrap errors from the 'pkg` tier with more user friendly message when available.  Not blocking for this PR.

Reply:
> Can you help clarify on the goals of the proposal? Is it:
> 
> 1. To reshare implementation for error formatting
> 2. To decouple ActionResult from the error
> 3. To provide the opportunity for ux.Error to perform `MarshalJson`?

All of above.
I see a world where actions.ActionResult can wrap a success or error result, but it should not define how the error or success should look in the console.
The actions.ActionResult has a method `ToUxItem()` which produced ux.ActionResult.
I think we should break `ux.ActionResult` into `ux.SuccessResult` and `ux.ErrorResult`.   That would allow us to define specific style for each and use one of the another

Reply:
This is now irrelevant given the revert, but I would like to see concrete real-world examples that exemplify the usefulness illustrated. I believe we're over-burdening and centralizing core logic into a "ux" layer that could otherwise be abstracted into a proper library that isn't "ux" specific.

I also believe that the JSON presentation logic of the cli requires revisitation, and that will end up impacting whatever short-term design we've created.